doc/md/guides/mysql/cd-101.md

```diff
@@ -0,0 +1,156 @@
+
+This document describes how Atlas's linter can be used to warn about dropping
+foreign key constraints from a MySQL database.
+
+Let's demonstrate by creating a database with 2 tables - users and pets:
+```sql
+CREATE TABLE users (
+	    id bigint NOT NULL AUTO_INCREMENT,
+	    name varchar(255) NOT NULL,
+	    PRIMARY KEY (id)
+	);
+
+CREATE TABLE pets (
+        id bigint NOT NULL AUTO_INCREMENT,
+        name varchar(255) NOT NULL,
+        owner_id bigint,
+        PRIMARY KEY (id),
+        CONSTRAINT owner_id FOREIGN KEY (owner_id) REFERENCES users (id)
+    );
+```
+
+Since we declared the foreign key constraint on the pets table,
+the database will validate any data inserted to it - blocking insertion
+of rows where the `owener_id` does not correspond to an `id` of an existing user.
+
+Now, let's add a new migration, dropping this constraint as described in the
+[MySQL documentation](https://dev.mysql.com/doc/refman/8.0/en/create-table-foreign-keys.html#:~:text=the%20foreign%20key.-,Dropping%20Foreign%20Key%20Constraints,-You%20can%20drop)
+:
+```sql
+ALTER TABLE pets DROP FOREIGN KEY owner_id;
+```
+
+This action, while won't cause data loss, can have some unpredictable side effects
+(as described below). This is why we may want to run some inspection to provide us 
+with some alerts when attempting to drop a foreign-key constraint.
+This can be achieved with Atlas's linter by running:
+
+```bash
+atlas migrate lint --dir file://migrations --dev-url "mysql://root:pass@localhost:3306/dev" --latest 1
+```
+
+In our example, the last command will produce the following output:
+```bash
+20230305125128.sql: constraint deletion detected:
+
+	L1: Dropping foreign-key constraint "owner_id"
+```
+
+([Click here](/versioned/lint) for more information about how to use the atlas cli linter).
+
+## Impact of dropping foreign key constraints
+
+Like mentioned above, dropping a foreign key constraint from a table is not a destructive
+action, meaning it won't cause data loss. And yet, it can have some negative impact since 
+it influences the business logic of our system.
+
+### Data integrity
+
+Let's imagine we have a service which exposes a REST API on top of our database.
+Our service has an endpoint in `/pets/{id}` which is used to fetch the name and 
+the name of the owner of a pet, given its ID.
+When calling this endpoint, the application is executing the following JOIN query:
+
+```sql
+SELECT
+    pets.id,
+    pets.name,
+    users.name AS owner_name
+FROM pets JOIN users ON pets.owner_id = users.id
+WHERE pets.id = {id};
+```
+
+Given an existing pet ID, our service (and it's consumers) may assume that this
+query won't be empty. This is because the foreign key constraint makes sure that
+the `owner_id` field will point to an existing row in the users table. But if we
+remove the constraint, we may find some pets with invalid owner IDs. Therefor, the 
+query may return empty, potentially causing a failure in the application.
```

Comment:
This is not 100% correct as the table definition does not state a NOT NULL on the owner_id which means there still can be a pet without an owner.